rdflib/namespace.py

```diff
@@ -123,28 +118,27 @@ def __new__(cls, value):
 
     @property
     def title(self):
+        # Override for DCTERMS.title to return a URIRef instead of str.title method
         return URIRef(self + "title")
 
     def term(self, name):
         # need to handle slices explicitly because of __getitem__ override
         return URIRef(self + (name if isinstance(name, str) else ""))
 
-    def __getitem__(self, key, default=None):
+    def __getitem__(self, key):
         return self.term(key)
 
     def __getattr__(self, name):
         if name.startswith("__"):  # ignore any special Python names!
             raise AttributeError
-        else:
-            return self.term(name)
+        return self.term(name)
 
     def __repr__(self):
```

Comment:
I may be wrong here, but AFAIK `f"Namespace({self!r})"` will call `self.__repr__`,  if this is the case it will result in infinite recursion.

Reply:
I guess just doing `return f"Namespace({self})"` should be fine and do what was happening before.

Reply:
This `f"Namespace({self!r})"` is fine as long as `__str__` is implemented. Since we are subclassing `Namespace` from `str`, this is fine and does not result in an infinite recursion.

Reply:
I can't see any issues with `return f"Namespace({self})"` though either. Maybe @kernc can explain further?

Reply:
Just checked:

```
20210707T222944 iwana@teekai.zoic.eu.org:~/d/github.com/iafork/rdflib
$ git diff
diff --git a/test/test_namespace.py b/test/test_namespace.py
index 59ff0ece..92b9e495 100644
--- a/test/test_namespace.py
+++ b/test/test_namespace.py
@@ -155,3 +155,8 @@ class NamespacePrefixTest(unittest.TestCase):
 
         ref = URIRef('http://www.w3.org/1999/02/22-rdf-syntax-ns#type')
         self.assertTrue(ref in RDF, "_RDFNamespace does not include rdf:type")
+
+    def test_repr(self):
+        ns_str = "http://example.com/name/space"
+        ns = Namespace(ns_str)
+        self.assertEqual(ns_str, f"{ns!r}")
20210707T222949 iwana@teekai.zoic.eu.org:~/d/github.com/iafork/rdflib
$ ^C
20210707T223015 iwana@teekai.zoic.eu.org:~/d/github.com/iafork/rdflib
$ poetry run python run_tests.py --tests test/test_namespace.py 
Running nose with: --tests test/test_namespace.py --attr= --where=./ --with-doctest --doctest-extension=.doctest --doctest-tests
.....E....
======================================================================
ERROR: test_repr (test.test_namespace.NamespacePrefixTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/iwana/d/github.com/iafork/rdflib/test/test_namespace.py", line 162, in test_repr
    self.assertEqual(ns_str, f"{ns!r}")
  File "/home/iwana/d/github.com/iafork/rdflib/rdflib/namespace.py", line 144, in __repr__
    return f"Namespace({self!r})"
  File "/home/iwana/d/github.com/iafork/rdflib/rdflib/namespace.py", line 144, in __repr__
    return f"Namespace({self!r})"
  File "/home/iwana/d/github.com/iafork/rdflib/rdflib/namespace.py", line 144, in __repr__
    return f"Namespace({self!r})"
  [Previous line repeated 235 more times]
RecursionError: maximum recursion depth exceeded while calling a Python object

----------------------------------------------------------------------
Ran 10 tests in 0.005s

FAILED (errors=1)
```

Reply:
The `!r` at the end there means it will call __repr__ and not str, which is the function itself.

Will make a PR to the branch.

Reply:
Please merge: https://github.com/kernc/rdflib/pull/1

Reply:
@aucampia, @edmondchuc 
[`__repr__` output](https://docs.python.org/3/reference/datamodel.html#object.__repr__):
> should look like a valid Python expression that could be **used to recreate an object with the same value**

So one of these would be better, I think:
```py
return f"Namespace({super().__repr__()})"
return f"Namespace('{self}')"  # assuming no quotes in namespace URI
```

Reply:
@aucampia Nice, thanks for the check.

Reply:
@kernc I updated PR with your suggestion and added tests to validate.